util/patch/options.go

```diff
@@ -30,11 +30,26 @@ type HelperOptions struct {
 	// on the incoming object to match metadata.generation, only if there is a change.
 	IncludeStatusObservedGeneration bool
 
+	// ForceOverwriteConditions allows the patch helper to overwrite conditions in case of conflicts.
+	// This option should only ever be set in controller managing the object being patched.
+	ForceOverwriteConditions bool
+
 	// OwnedConditions defines condition types owned by the controller.
 	// In case of conflicts for the owned conditions, the patch helper will always use the value provided by the controller.
+	//
+	// DEPRECATED: Use ForceOverwriteConditions.
```

Comment:
Should we deprecate also the `func WithOwnedConditions`?

Reply:
I was thinking maybe we should just keep both? It could allow different use cases, although if we prefer to deprecate both that's fine as well

Reply:
There is already a lot of complexity around patching conditions, so I am for deprecating both and, unless a concrete use case pos ups soon, remove.